src/main/scala/scalismo/statisticalmodel/DiscreteLowRankGaussianProcess.scala

```diff
@@ -220,7 +220,7 @@ case class DiscreteLowRankGaussianProcess[D <: Dim: NDSpace, DO <: Dim: NDSpace]
    * Interpolates discrete Gaussian process to have a new, continuous representation as a [[DiscreteLowRankGaussianProcess]],
    * using nearest neigbor interpolation (for both mean and covariance function)
    */
-  override def interpolateNearestNeighbor: LowRankGaussianProcess[D, DO] = {
+  override def interpolateNearestNeighbor: NearestNeighbourInterpolatedLowRankGaussianProcess[D, DO] = {
```

Comment:
This should not return a NearestNeighborInterpolatedLowRankGaussianProcess, but a LowRankGaussianProcess, as the former is a private class and should not be exposed in the public API (I am surprised that you don't get a warning from scala).